library/Zend/Db/Adapter/Pdo/Abstract.php

```diff
@@ -311,15 +313,20 @@ protected function _beginTransaction()
     protected function _commit()
     {
         $this->_connect();
-        $this->_connection->commit();
+        if (!$this->_connection->commit()) {
+            throw new Zend_Db_Adapter_Exception('Commit failed (due to deadlock for Galera Cluster)');
```

Comment:
asi bych tam nepsal nic o galera clusteru, když to může v budoucnosti běžet i jinde
stejně tak tohle je úplně abstraktní adaptér třeba pro postgre

Reply:
mysql mimo galeru by tu nikdy failnout nemělo, jiné databáze nepoužíváme, ale AFAIK by v žádné databázi neměl commit nikdy failnout :)
dal jsem to do závorky proto, abych to bylo hned zřejmé a každý nemusel zjišťovat, jak je možné, že commit failne, když by přece nikdy failnout neměl :D
proto bych tady tu zmínku nechal

Reply:
no ideální to není, vhodnější by bylo detekovat že fakt běžíme na galera clusteru nebo mít přetíženou metodu z pdo_mysql adaptéru, ale jsme poslední firma na světě udržující ZF1, tak budiž...